pantheon/src/main/java/tech/pegasys/pantheon/cli/operator/OperatorSubCommand.java

```diff
@@ -0,0 +1,322 @@
+
+    /**
+     * Computes RLP encoded exta data from pre filled list of addresses.
+     *
+     * @throws IOException
+     */
+    private void processExtraData() throws IOException {
+      final JsonGenesisConfigOptions genesisConfigOptions =
+          JsonGenesisConfigOptions.fromJsonObject(genesisConfig.getJsonObject("config"));
+      if (genesisConfigOptions.isIbft2() || genesisConfigOptions.isIbftLegacy()) {
+        LOG.info("Generating IBFT extra data.");
+        final String extraData =
+            IbftExtraData.fromAddresses(addressesForGenesisExtraData).encode().toString();
+        genesisConfig.put("extraData", extraData);
+      }
+    }
+
+    private void createFileAndWrite(
+        final Path directory, final String fileName, final String content) throws IOException {
+      final Path filePath = directory.resolve(fileName);
+      Files.createFile(filePath);
+      Files.write(filePath, content.getBytes(UTF_8));
+    }
+
+    /**
+     * Parses the root configuration file and related sub elements.
+     *
+     * @throws IOException
+     */
+    private void parseConfig() throws IOException {
+      final String configString =
+          Resources.toString(configurationFile.toPath().toUri().toURL(), UTF_8);
+      operatorConfig = new JsonObject(configString);
+      genesisConfig = operatorConfig.getJsonObject("genesis");
+      blockchainConfig = operatorConfig.getJsonObject("blockchain");
+      nodesConfig = blockchainConfig.getJsonObject("nodes");
+      generateNodesKeys = nodesConfig.getBoolean("generate", false);
+    }
+
+    /**
+     * Checks if the output directory exists. If forceOverwrite the output directory will be
+     * flushed.
+     *
+     * @throws IOException
+     */
+    private void handleOutputDirectory() throws IOException {
+      checkNotNull(outputDirectory);
+      final Path outputDirectoryPath = outputDirectory.toPath();
+      if (outputDirectory.exists()
+          && outputDirectory.isDirectory()
+          && outputDirectory.list() != null
+          && outputDirectory.list().length > 0) {
+        throw new IllegalArgumentException("Output directory must be empty.");
+      } else if (!outputDirectory.exists()) {
+        Files.createDirectory(outputDirectoryPath);
+      }
+      keysDirectory = outputDirectoryPath.resolve("keys");
+      Files.createDirectory(keysDirectory);
+    }
+
+    /**
+     * Write the content of the genesis to the output file.
+     *
+     * @param directory The directory to write the file to.
+     * @param fileName The name of the output file.
+     * @param genesis The genesis content.
+     * @throws IOException
+     */
+    private void writeGenesisFile(
+        final File directory, final String fileName, final JsonObject genesis) throws IOException {
+      final Path genesisPath = directory.toPath().resolve(fileName);
+      if (Files.exists(genesisPath)) {
+        throw new IllegalArgumentException("Output genesis file must not exist.");
```

Comment:
Can you add a test that hits this?

Reply:
Actually it is not possible, the genesis file is created under the output directory. If output directory exists it would have fail earlier. This is a newly created directory. I remove this check.